### ISSUE TITLE
OpenSSL build_apps target fails due to unresolved references

### ISSUE THREAD

**Issue by mastacheata** [SynoCommunity/spksrc#2768]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ tvheadend
_Package Version:_ 4.2.1 
The actual problem is with openssl-1.0.2k, though.

_NAS Model:_ 713+
_NAS Architecture:_ cedarview
_DSM version:_ 6.1
I tried to build this in the Docker image supplied in this repository.

### Expected behavior
The openSSL libraries should be built and in turn the tvheadend package

### Actual behavior
The openSSL libraries fail to build due to undefined references and in turn the whole package build fails

### Steps to reproduce
_1._ Pick up any package with openssl dependency (in this case I tried tvheadend 4.2.1, i.e. the latest version available at the time of writing and before m4tt075's PR for 4.2.2 gets merged.
_2._ wait for it to break havoc after a couple of minutes (on the first run or immediately on the second-nth run due to the previous subbuilds successfully finishing) 

#### Other logs
Here's a gist with the 7000 line build of tvheadend .
The relevant part begins here on line 3087: https://gist.github.com/mastacheata/a8a4b8c6cf6e067f45c3371848f4c814#file-conemu-2017-05-16-p9252-log-L3087

This is where it starts building the build_apps subtarget of openssl.

From what I could find on the internet this might be caused by trying to build openssl with multiple jobs in parallel. However running make with make -j1, effectively turning off the job scheduler and thus parallel builds doesn't fix the problem.
It still fails at the exact same position with the exact same errors. (obviously the position is only the same if you do a `make clean` first, since it will skip the already successful build targets otherwise)

**Comment by m4tt075** [comment]
I cannot reproduce this on Virtualbox, but have no experience with docker whatsoever. I'm therefore afraid I cannot help.

I know that @dreamcat4 has been building docker images for TVH though. You can find his work [here](https://hub.docker.com/r/dreamcat4/tvheadend/). Seems a little bit outdated, but in the comments section on that website someone outlines how to get the release image.

Does not answer your question though...

**Comment by mastacheata** [comment]
Whoa, we're on the road to a major misunderstanding here:
What @dreamcat4 does is RUN a precompiled binary of tvheadend in Docker.
What I tried was building/compiling the tvheadend sources into a binary package for the Diskstation.

Two very different things!

**Comment by m4tt075** [comment]
Right. I always thought docker images were platform-agnostic, which apparently they are not.

**Comment by ymartin59** [comment]
With Docker image from https://hub.docker.com/r/synocommunity/spksrc/ running on a x86_64 Linux system, I have run `make arch-cedarview-5.2` to build `tvheadend_cedarview-5.2_4.2.1-8.spk` with latest `master` and `openssl-1.0.2k` is included there. Same result after `make clean` and `make -j4 arch-cedarview-5.2`.

You seem interested by a package for DSM 6.1 - so I propose to build with corresponding toolchain: use PR #2727 source branch https://github.com/ymartin59/spksrc/tree/dsm6-toolchains-6.1

I have just produced successfully package `tvheadend_cedarview-6.1_3.4.27-6.spk` with target `arch-cedarview-6.1` and `openssl-1.0.2k` is OK too. Version 3.4.27 comes from `dsm6` branch where recent `tvheadend` update on `master` has not been merged into yet.

After merging both recent master and DSM 6.1 toolchains branch, I produced `tvheadend_cedarview-6.1_4.2.1-8.spk` without troubles.

So my best guess - after finding toolchain checksum error at the beginning of your console output - is that a previous failure has left some troubles in your working copy (or maybe Docker image/container even if I doubt about it). I propose you do a fresh `git clone` and stop your container, remove image and restart container with a freshly downloaded image, then goes to `/spksrc/tvheadend` to run `make arch-cedarview` - it should really work.

**Comment by mastacheata** [comment]
How exactly do I get around the checksum error again?
IIRC all you had to do was run the sha1sum on the .wrong file, fix the digest with the right checksum, make clean, make again and your good, but that doesn't seem to work for me at the moment...

**Comment by ymartin59** [comment]
I recommend a fresh `git clone` because `make clean` does not remove downloaded archives in `distrib`, nor working directory of each toolchain in `toolchains/syno-ARCH-x.y/work` where any download/network or extraction/out-of-space may contain inconsistent files/states and so are left unfix.

**Comment by m4tt075** [comment]
@mastacheata I have fixed some problems with docker builds recently and remembered your issue. I have just successfully built `cedarview-5.2` using `docker` via my `travis-ci` interface. [Here is the result](https://travis-ci.org/m4tt075/spksrc/builds/326961718). If your issue is still relevant, please try again. If not, or the issue has been solved otherwise for you in the meantime, please close this issue. Thank you.

**Comment by Safihre** [comment]
The `cross/openssl` now has the right version and digests to compile many many packages 👍
